src/meta/processors/zone/MergeZoneProcessor.cpp

```diff
@@ -8,7 +8,235 @@
+      auto zoneValueRet = doGet(std::move(zoneKey));
+      auto hosts = MetaKeyUtils::parseZoneHosts(std::move(nebula::value(zoneValueRet)));
+      for (auto& host : hosts) {
+        auto hp = hostParts.find(host);
+        if (hp == hostParts.end()) {
+          LOG(ERROR) << "Host " << host << " not found";
+          code = nebula::cpp2::ErrorCode::E_NO_HOSTS;
+          break;
+        }
+
+        auto& parts = hp->second;
+        for (auto part : parts) {
+          auto it = std::find(totalParts.begin(), totalParts.end(), part);
+          if (it != totalParts.end()) {
+            LOG(ERROR) << "Part " << part << " have exist";
+            code = nebula::cpp2::ErrorCode::E_CONFLICT;
+            break;
+          }
+        }
+
+        CHECK_CODE_AND_BREAK();
+        totalParts.insert(totalParts.end(), parts.begin(), parts.end());
+      }
+      CHECK_CODE_AND_BREAK();
+    }
+    CHECK_CODE_AND_BREAK();
+    iter->next();
+  }  // space
+
+  if (code != nebula::cpp2::ErrorCode::SUCCEEDED) {
+    LOG(ERROR) << "Check parts failed, error " << apache::thrift::util::enumNameSafe(code);
+    handleErrorCode(code);
+    onFinished();
+    return;
+  }
+
+  // Rewrite space properties
+  ret = doPrefix(spacePrefix);
+  iter = nebula::value(ret).get();
+  std::vector<kvstore::KV> data;
+  while (iter->valid()) {
+    auto spaceKey = iter->key();
+    auto properties = MetaKeyUtils::parseSpace(iter->val());
+    auto spaceZones = properties.get_zone_names();
+    bool replacement = false;
+    for (auto& zone : zones) {
+      auto it = std::find(spaceZones.begin(), spaceZones.end(), zone);
+      if (it != spaceZones.end()) {
+        LOG(INFO) << "REMOVE ZONE " << zone;
+        replacement = true;
+        spaceZones.erase(it);
+      }
+    }
+
+    if (replacement) {
+      spaceZones.emplace_back(zoneName);
+      properties.set_zone_names(std::move(spaceZones));
+      auto spaceVal = MetaKeyUtils::spaceVal(properties);
+      data.emplace_back(std::move(spaceKey), std::move(spaceVal));
+    }
+    iter->next();
+  }
+
+  // Write the merged zone into meta service
+  auto key = MetaKeyUtils::zoneKey(zoneName);
+  std::vector<HostAddr> zoneHosts;
+  for (auto& zone : zones) {
+    auto zoneKey = MetaKeyUtils::zoneKey(zone);
+    auto zoneValueRet = doGet(std::move(zoneKey));
+    auto hosts = MetaKeyUtils::parseZoneHosts(std::move(nebula::value(zoneValueRet)));
+    zoneHosts.insert(zoneHosts.end(), hosts.begin(), hosts.end());
+  }
+  auto zoneVal = MetaKeyUtils::zoneVal(std::move(zoneHosts));
+
+  // Remove original zones
+  LOG(INFO) << "Remove original zones size " << zones.size();
+  std::vector<std::string> zoneKeys;
+  for (auto& zone : zones) {
+    LOG(INFO) << "Remove Zone " << zone;
+    zoneKeys.emplace_back(MetaKeyUtils::zoneKey(zone));
+  }
+  folly::Baton<true, std::atomic> baton;
+  auto result = nebula::cpp2::ErrorCode::SUCCEEDED;
+  kvstore_->asyncMultiRemove(kDefaultSpaceId,
```

Comment:
Why don't use LogBatch? You can't promise it is atomic. So as `DropHost` and other processor, please check it

Reply:
Check all processor zone related, e.g. `DropHostProcessor` in previous PR